api/app/models/resource.rb

```diff
@@ -90,6 +94,12 @@ def resource_to_event
     )
   end
 
+  def queue_fetch_thumbnail
+    return unless external_id_changed? || new_record?
+    return if variant_thumbnail.present? && !external_id_changed?
```

Comment:
Your conditions here are too restrictive. They make it impossible, for example, to trigger a thumbnail fetch for a record that existed before this featured was added. You also don't check whether the resource is an external_video in this method. You're relying on the if condition in the after_save callback, but that's insufficient. This `queue_fetch_thumbnail` is a public method on the model and could be called from anywhere in the application — what happens if its called on something that's not a video?

The logic here needs to be made more granular. I'd replace this with something like this:
```
  def queue_fetch_thumbnail
    return unless external_video?
    if !variant_thumbnail.present? || external_id_changed?
      FetchVideoThumbnails.perform_later(id)
    end
  end
```

This is more to the point of what we really want, which is to fetch thumbnails for videos that have changed or for videos that should have a thumbnail but do not.

This change will introduce a different problem, however. If the thumbnail fetch fails for some reason, we'll be caught in an infinite loop in which the video saves without a thumbnail and a new job is spawned. The only way to work around this is to save a record of the failed attempt. I would track this in a separate model, so as not to pollute the resource with unrelated concerns. Make a model called "thumbnail_fetch_attempts" and store the resource, the attempted URL, success boolean, and attempts. We can use this to determine whether an attempt should be queued.